### PR TITLE
Ability to add value to insight objects and update it #31

### LINKED ISSUE #31: Ability to add value to insight objects and update it

### PR DESCRIPTION
**Pull request by mtss92**

### CODE REVIEW

File: src/AtlassianAssistance.JiraService/Attributes/InsightObjectTypeAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace AtlassianAssistance.JiraService.Attributes
+{
+    public sealed class InsightObjectTypeAttribute : Attribute
```
Comment (kalhorim) [inline src/AtlassianAssistance.JiraService/Attributes/InsightObjectTypeAttribute.cs:7]: Please add notation for use just in class.

File: src/AtlassianAssistance.JiraService/Attributes/InsightAttributeFieldAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace AtlassianAssistance.JiraService.Attributes
+{
+    public sealed class InsightAttributeFieldAttribute : Attribute
```
Comment (kalhorim) [inline src/AtlassianAssistance.JiraService/Attributes/InsightAttributeFieldAttribute.cs:7]: Please add a notation to use just in properties.

File: src/AtlassianAssistance.JiraService/Services/SchemaService.cs [hunk 3]
```diff
@@ -36,6 +42,98 @@ public async Task<IEnumerable<T>> GetFieldValues<T>(CustomFieldValuesRequest<T>
...
+            dynamic myObject = new { objectTypeId = objectTypeId, attributes = objAttributes.ToArray() };
+            return myObject;
+        }
+
+        private async Task<InsightObjectAttribute> CreateObjectAttribute(System.Reflection.PropertyInfo prop, object value)
+        {
+            var attribute = new InsightObjectAttribute();
+            attribute.objectTypeAttributeId = (prop.GetCustomAttributes(typeof(InsightAttributeFieldAttribute), false).First() as InsightAttributeFieldAttribute).ObjectTypeAttributeId;
+
+            if (value is JiraCustomFieldBase customField)
```
Comment (kalhorim) [inline src/AtlassianAssistance.JiraService/Services/SchemaService.cs:100]: Please doesn't use too many if statement.

**Review by kalhorim — CHANGES_REQUESTED** [review]
Before applying changes, we can make a call for reviewing process.

File: src/AtlassianAssistance.JiraService/Attributes/InsightObjectTypeAttribute.cs (hunk 1 above)
Comment (mtss92) [reply src/AtlassianAssistance.JiraService/Attributes/InsightObjectTypeAttribute.cs:8]: [5c71fbf](https://github.com/kalhorim/JiraService/commit/5c71fbf7addc091cfeb99bea1cdd44d7b997b18b)

File: src/AtlassianAssistance.JiraService/Attributes/InsightAttributeFieldAttribute.cs (hunk 2 above)
Comment (mtss92) [reply src/AtlassianAssistance.JiraService/Attributes/InsightAttributeFieldAttribute.cs:8]: [94dfb5e](https://github.com/kalhorim/JiraService/commit/94dfb5ec1c0052e78cdb631c9081becb8f45b00a)

File: src/AtlassianAssistance.JiraService/Services/SchemaService.cs (hunk 3 above)
Comment (mtss92) [reply src/AtlassianAssistance.JiraService/Services/SchemaService.cs]: [b3e49f4](https://github.com/kalhorim/JiraService/pull/32/commits/b3e49f45f9ef9f427ecb6bdc4b7195c1531fbdd5)

**Review by kalhorim — APPROVED** [review]
Great!

**Pull request merged**